lib/datadog/tracing/distributed/trace_context.rb

```diff
@@ -309,9 +314,9 @@ def extract_datadog_fields(dd_tracestate)
           [origin, sampling_priority, tags, unknown_fields]
         end
 
-        # Restore `:` back to `=`.
+        # Restore `~` back to `=`.
         def deserialize_tag_value(value)
-          value.tr!(':', '=')
+          value.tr!('~', '=')
           value
```

Comment:
I'm curious -- why use `tr!` instead of `gsub!`? Is it a performance optimization?

Reply:
`tr!` is much faster for single character replacement:
```ruby
require 'benchmark/ips'

Benchmark.ips do |x|
  x.report('gsub!') { 'a'.gsub!('a', 'b') }
  x.report('tr!') { 'a'.tr!('a', 'b') }
  x.compare!
end

# ruby 3.2.1
# Comparison:
#                  tr!:  3905918.3 i/s
#                gsub!:  1702250.5 i/s - 2.29x  slower
```